var/spack/repos/builtin/packages/parallel-netcdf/package.py

```diff
@@ -58,6 +60,7 @@ def url_for_version(self, version):
     variant("pic", default=True, description="Produce position-independent code (for shared libs)")
     variant("shared", default=True, description="Enable shared library")
     variant("burstbuffer", default=False, description="Enable burst buffer feature")
+    variant("examples", when="@1.13:", default=False, description="Install example programs")
```

Comment:
Technically, it is possible to install versions `@:12` without the examples but the `when="@1.13:"` argument doesn't allow for `parallel-netcdf@:1.12~examples`. Therefore, I would use a conflict:
```suggestion
    variant("examples", default=False, description="Install example programs")
    conflicts("+examples", when="@:1.12")
```
I am also curious about the value of the installed example programs. What is the use case when it makes sense?

Reply:
Thanks for the quick review @skosukhin. I'm on the stack maintainer side, so my perspective is just to get up-to-date versions of these libraries for our users/developers. As such, I don't have a specific use case, but I thought some of our developers may find the compiled examples useful as a reference to compare with the raw code in the source tarball/Git repo.

All that said, I'm happy to remove this variant if you think that makes sense, or accept your suggestion for a conflict otherwise (thanks for explaining the rationale).

Reply:
[This](https://github.com/Parallel-NetCDF/PnetCDF/pull/91) provides some information about the option. Basically, in some cases, it is possible to run the examples only after the installation. Let's have the variant then.

Reply:
Sounds good. I've committed the suggested change too.